app/components/login/NewUserRegistration.js

```diff
@@ -0,0 +1,93 @@
+//Placeholder for newuser registration component
+
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { createUser } from '../../redux/user';
+import TextField from '@material-ui/core/TextField';
+import Button from '@material-ui/core/Button';
+import { useStyles } from '../index';
+import { Link } from 'react-router-dom';
+
+function NewUserRegistration(props) {
+  const user = useSelector(state => state.user);
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const dispatch = useDispatch();
+
+  const classes = useStyles();
+
+  const handleChange = (event) => {
+    switch (event.target.id) {
+      case "firstname":
+        setFirstName(event.target.value);
+        break;
+      case "lastname":
+        setLastName(event.target.value);
+        break;
+        case "email":
+        setEmail(event.target.value);
+        break;
+        case "password":
+        setPassword(event.target.value);
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = () => {
+    dispatch(createUser({ firstName, lastName, email, password}));
```

Comment:
what happens if createUser fails?
The User model has a isUnique constraint on the email property. Either the thunk or the API call should accomodate for if creation fails because of this.

Reply:
yes its TODO as mentioned above but want this merged in the train of changes with small changes for below later: TODO: a) order, shipping update if guest created order and shipping address (i.e. did checkout) before deciding to register. This needs to be updated and synchronized. b) existing email validation. not done.